pkg/status/metrics.go

```diff
@@ -29,7 +29,7 @@ func init() {
 	stdprometheus.MustRegister(releaseCondition)
 }
 
-func ObserveReleaseConditions(old v1.HelmRelease, new *v1.HelmRelease) {
+func ObserveReleaseConditions(old *v1.HelmRelease, new *v1.HelmRelease) {
 	conditions := make(map[v1.HelmReleaseConditionType]*v1.ConditionStatus)
 
 	for _, condition := range old.Status.Conditions {
```

Comment:
Given the pointer change means that `old` can now be `nil` too, I would add an additional check to see if `old != nil` to prevent other panics from happening.